### ISSUE TITLE
Decouple project from Bootstrap 3

### ISSUE THREAD

**Issue by talvbansal** [talvbansal/media-manager#87]
I'm not sure how to achieve this but ideally it'd be good to find a way to decouple this from a particular UI framework but feel native to the project that its being used in.

Currently it's tied to bootstrap 3, I'm about to use it in a bootstrap 4 project and people like @partikule are trying to use it with bulma.

I imagine we'd have to make heavy use of scoped slots but it would also be nice to provide a default fallback UI which I'm not sure how to use either

Any help on this would be really appreciated

**Comment by partikule** [comment]
We're currently adapting it to Bulma and add / modify some features to enhance it.

What we do : 
- Decouple the media-manager from its parent modal : it should be possible to add the component in the modal of our choice.
- Remove the use of EventHub : this.$emit is enough for all main events. We will perhaps keep one call to one eventHub to make the main app able to react to an image upload / delete for example.

We will think about bootstrap decoupling but I agree this will ba a little bit tricky, as Bootstrap doesn't have the same tag logic when it comes to form elements.
Perhaps one solution could be to provide both views, or to have 2 versions sharing the same main code...

Anyway, I hope I will find the time to add our mods to our fork, in one dedicated branch.
I'll keep you informed.

**Comment by talvbansal** [comment]
@partikule thanks for your feedback - following your suggestions previously I've gone through and removed most of the eventHub use cases - the only one im not sure how to handle is that for the notifications. Do you have any idea on how we could handle those?

Can you share what you did to remove it from the main modal id be interested on how you've made it work so that it works in both instances.

Thanks

**Comment by partikule** [comment]
After some search, I find out it is possible.
... but it will be difficult to implement as it will be difficult to use for final users.

https://adamwathan.me/renderless-components-in-vuejs

**Comment by talvbansal** [comment]
Yeah I’ve been meaning to read this and I’ve signed up for the new course he’s doing.

My top level take away is that you’d In essence ship the functionality with no ui then get the implementing project to write a ui that is tailored for its needs. 

I’m thinking we could do the above but also provide a couple of preset examples that could be used to get people going in separate builds. Users can then choose to use the build for their framework or just build the whole ui from scratch.

I’m not sure how feasible it’ll be but if I can make that happen would you be willing to share your bulma implementation ?

**Comment by partikule** [comment]
Yeah, for sure I guess too this is the way... but the way is time consuming :-(

Btw, did you had a look at the sceencasts of the Bulma implementation I made ?

**Comment by talvbansal** [comment]
I haven’t can you give me a link to it please 😱

**Comment by partikule** [comment]
I've send the message to your personal email address [at]hotmail.com :-)

**Comment by partikule** [comment]
https://github.com/partikule/media-manager/tree/bulma

=> Bulma version, unfinished.
